service/src/main/java/uk/gov/hmcts/reform/fpl/controllers/support/MigrateCaseController.java

```diff
@@ -48,28 +49,45 @@ public AboutToStartOrSubmitCallbackResponse handleAboutToSubmit(@RequestBody Cal
         return respond(caseDetails);
     }
 
-
-    private void run2469(CaseDetails caseDetails) {
+    private void run2491(CaseDetails caseDetails) {
         CaseData caseData = getCaseData(caseDetails);
 
-        if ("SN20C50010".equals(caseData.getFamilyManCaseNumber())) {
+        if ("SN20C50018".equals(caseData.getFamilyManCaseNumber())) {
             List<Element<HearingBooking>> hearings = caseData.getHearingDetails();
 
-            int hearingIndex = 0;
-            LocalDate expectedHearingDate = LocalDate.of(2020, Month.OCTOBER, 14);
+            LocalDate expectedHearing1Date = LocalDate.of(2021, Month.FEBRUARY, 5);
+            LocalDate expectedHearing2Date = LocalDate.of(2021, Month.MARCH, 4);
+
+            if (hearings.size() < 5) {
+                throw new IllegalArgumentException(
+                    format("Case has %s hearing(s), expected at least %s", hearings.size(), 5));
+            }
+
+            Element<HearingBooking> hearing1 = hearings.get(3);
+            Element<HearingBooking> hearing2 = hearings.get(4);
 
-            if (hearings.size() < hearingIndex + 1) {
-                throw new IllegalArgumentException(format("Case %s has %s hearing(s), expected at least %s ",
-                    caseDetails.getId(), hearings.size(), hearingIndex + 1));
+            if (!expectedHearing1Date.equals(hearing1.getValue().getStartDate().toLocalDate())) {
+                throw new IllegalArgumentException(
+                    format("Invalid hearing date %s", hearing1.getValue().getStartDate().toLocalDate()));
             }
 
-            LocalDate hearingDate = hearings.get(hearingIndex).getValue().getStartDate().toLocalDate();
+            if (!expectedHearing2Date.equals(hearing2.getValue().getStartDate().toLocalDate())) {
+                throw new IllegalArgumentException(
+                    format("Invalid hearing date %s", hearing2.getValue().getStartDate().toLocalDate()));
+            }
+
+            if (!HearingType.FURTHER_CASE_MANAGEMENT.equals(hearing1.getValue().getType())) {
+                throw new IllegalArgumentException(
+                    format("Invalid hearing type %s", hearing1.getValue().getType()));
+            }
 
-            if (!expectedHearingDate.equals(hearingDate)) {
-                throw new IllegalArgumentException(format("Invalid hearing date %s", hearingDate));
+            if (!HearingType.ISSUE_RESOLUTION.equals(hearing2.getValue().getType())) {
+                throw new IllegalArgumentException(
+                    format("Invalid hearing type %s", hearing2.getValue().getType()));
             }
 
-            hearings.remove(hearingIndex);
+            hearings.remove(hearing2);
```

Comment:
If they are duplicate will this remove the expected one? By being duplicate it will remove based on `equals` method but this will be the same for 2 hearings. You are getting the 4th and 5th hearing to assert for removal but from the test it seems that it determines that 2 earlier ones are removed. Maybe it would be safer to remove on index here.

Reply:
ahhh no ignore me, the element means that it will be unique